Source/hydro/Castro_ctu_hydro.cpp

```diff
@@ -176,22 +176,37 @@ Castro::construct_ctu_hydro_source(Real time, Real dt)
 
       // compute the flattening coefficient
 
+      Array4<Real const> const q_arr = q.array(mfi);
       Array4<Real> const flatn_arr = flatn.array();
+#ifdef RADIATION
+      Array4<Real> const flatg_arr = flatg.array();
+#endif
 
       if (first_order_hydro == 1) {
         AMREX_PARALLEL_FOR_3D(obx, i, j, k, { flatn_arr(i,j,k) = 0.0; });
       } else if (use_flattening == 1) {
+
+        uflatten(obx, q_arr, flatn_arr, QPRES);
+
 #ifdef RADIATION
-        ca_rad_flatten(ARLIM_3D(obx.loVect()), ARLIM_3D(obx.hiVect()),
-                       BL_TO_FORTRAN_ANYD(q[mfi]),
-                       BL_TO_FORTRAN_ANYD(flatn),
-                       BL_TO_FORTRAN_ANYD(flatg));
-#else
-#pragma gpu box(obx)
-        ca_uflatten(AMREX_INT_ANYD(obx.loVect()), AMREX_INT_ANYD(obx.hiVect()),
-                    BL_TO_FORTRAN_ANYD(q[mfi]),
-                    BL_TO_FORTRAN_ANYD(flatn), QPRES+1);
+        uflatten(obx, q_arr, flatg_arr, QPTOT);
+
+        AMREX_PARALLEL_FOR_3D(obx, i, j, k,
+        {
+          flatn_arr(i,j,k) = flatn_arr(i,j,k) * flatg_arr(i,j,k);
+
+          if (Radiation::flatten_pp_threshold > 0.0) {
```

Comment:
We cannot directly read class member variables in the lambdas on the device. For now we'll need to make a local copy of this variable outside the parallel for.